src/preact/slot.js

```diff
@@ -129,8 +132,8 @@ export function useSlotContext(ref, opt_props) {
 }
 
 /**
- * @param {!Element} slot
- * @param {function(!AmpElement):void|function(!Array<!AmpElement>):void} action
+ * @param {HTMLSlotElement} slot
+ * @param {function(Element[] | Element):void|function(AmpElement[]):void} action
```

Comment:
Current param (I think): function accepting one or more elements and returning (either void OR a function accepting an array of amp-elements returning void)
```suggestion
 * @param {(function(Element[]|Element):void)|(function(AmpElement[]):void)} action
```

Reply:
I think a simpler construction would end up being `function(Element|Element[]|AmpElement[])`

Reply:
Nope, this is a tricky distinction:
`@param function(number)|function(string)` accepts either a function taking a string OR a function taking a number, but won't accept a function taking a (string or a number)
`@param function(number|string)` accepts either a function taking a (string or number), but won't accept a function that takes only one or the other

Reply:
Something something structural typing.

Based on the consumer, we can actually narrow this even more.

```
 * @param {function(Element|Element[]):void} action
```